### DISCUSSION TITLE
Switch linters from awesome-lint to markdownlint

### DISCUSSION THREAD

**Discussion by TyceHerrman** [APA-Technology-Division/urban-and-regional-planning-resources#77] [Ideas]
markdownlint seems more comprehensive than awesome-lint. For instance, markdownlint catches some blank lines inconsistencies in the existing Readme.

**Comment by d-wasserman** [comment]
I don't think we can do this per se without being in line with the requirements of the awesome-lists. It might be a yes and situation. I don't know if we want to completely switch. 
Would it increase actions processing significantly to use markdownlint? Do you know if the awesome-lists standards allow it as an alternative?

**Reply by TyceHerrman** [reply]
I don't think markdownlint is an alternative within awesome-lint. It does look like it encompasses all the linting of awesome-lint, but also has additional checks (like the number and location of blank lines). There's a well maintained Github action for markdownlint here for review/use https://github.com/marketplace/actions/markdownlint-mdl-action

GH Actions are free for public repos - I don't think markdownlint would be orders of magnitude different from awesome-lint, but even if it was, wouldn't it still be free as it's a public repo?

**Reply by d-wasserman** [reply]
I think it is not the free aspect I am concerned about, it is additional processing time to check contributions validation, and I don't want an action that duplicates some of what awesome-lint does. 
We currently use awesome-lint and a link validator. The link validator takes the most time, and adding another action feels like overkill for the repo. I think if we threw out awesome-lint that would put us out of step with the awesome-project, and if we add it can we deactivate the duplicative functionality?

**Reply by TyceHerrman** [reply]
Understood. 

Markdownlint allows for rule subset selection, so anything awesome-lint already does could be deactivated. I'll work on a Github Action that only runs markdownlint rules not already applied with awesome-lint for review.

**Reply by d-wasserman** [reply]
I would review and consider such an edit to our current actions in a PR. Thank you for taking an interest.
